src/container/mqtt.jl

```diff
@@ -0,0 +1,147 @@
+export MQTTProtocol,
+    send,
+    init,
+    close,
+    id,
+    has_message,
+    get_messages_channel,
+    disconnect,
+    subscribe
+
+using Mosquitto
+using Sockets: InetAddr
+
+# TODO TLS and connection handling for that
+
+mutable struct MQTTProtocol <: Protocol{String}
+    client::Client
+    broker_addr::InetAddr
+    connected::Bool
+    msg_channel::Channel
+    conn_channel::Channel
+    topic_to_aid::Dict{String,Vector{String}}
+
+    function MQTTProtocol(client_id::String, broker_addr::InetAddr)
+        # Have to cast types for the MQTT client constructor.
+        c = Client(string(broker_addr.host), Int64(broker_addr.port); id=client_id)
+        msg_channel = get_messages_channel(c)
+        conn_channel = get_connect_channel(c)
+        return new(c, broker_addr, false, msg_channel, conn_channel, Dict{String,Vector{String}}())
+    end
+end
+
+"""
+Initialize the Mosquitto looping task. This checks for incoming messages and 
+forwards their content to the container.
+
+# Returns
+
+Created Tasks
+"""
+function init(protocol::MQTTProtocol, stop_check::Function, data_handler::Function)
+    tasks = []
+    listen_task = errormonitor(
+        Threads.@spawn begin
+            try
+                Mosquitto.loop_start(protocol.client)
+
+                # listen for incoming messages and run callback
+                while true
+                    # handle incoming content messages
+                    while !isempty(protocol.msg_channel)
```

Comment:
I feel two whiles in outer while in try/catch is a bit much complexity, refactor into some nice functions